libopenarc/arc.c

```diff
@@ -2507,25 +2508,35 @@ arc_eoh(ARC_MESSAGE *msg)
 	}
 
 	/*
-	**  Request specific canonicalizations we want to run.
+	** Request specific canonicalizations we want to run.
 	*/
 
-	/* headers, validation */
 	h = NULL;
 	htag = NULL;
 	if (nsets > 0)
 	{
+		/* headers, validation */
 		h = msg->arc_sets[nsets - 1].arcset_ams;
 		htag = arc_param_get(h->hdr_data, "h");
-
 		if (strcmp(arc_param_get(h->hdr_data, "a"), "rsa-sha1") == 0)
 			hashtype = ARC_HASHTYPE_SHA1;
 		else
 			hashtype = ARC_HASHTYPE_SHA256;
 
-		status = arc_add_canon(msg, ARC_CANONTYPE_HEADER, msg->arc_canonhdr,
+		status = arc_parse_canon_t(arc_param_get(h->hdr_data, "c"),
```

Comment:
This may not be there; "c=" is not required (defaults to "simple/simple"; RFC6376 Section 3.5).